### PR TITLE
fix(rds): allow cluster from snapshot to enable encrypted storage

### LINKED ISSUE #17241: (aws-rds): DatabaseClusterFromSnapshot construct should support storage encryption
### General Issue

DatabaseClusterFromSnapshot construct should support storage enryption

### The Question

I need to enable encryption on an Aurora database cluster that already exists, and it appears the way to do this is to take a snapshot, and restore it to a new database cluster with the Storage Encrypted option enabled (and an encryption key specified):

> You can't convert an unencrypted DB cluster to an encrypted one. However, you can restore an unencrypted snapshot to an encrypted Aurora DB cluster. To do this, specify a KMS key when you restore from the unencrypted snapshot.

I've verified that I can do this manually through the AWS console. I also see that this `StorageEncrypted` option exists for the underlying CFN resource `AWS::RDS::DBCluster`, and also for the non-snapshotted `DatabaseCluster` CDK construct. **However**, the option for the CDK construct `DatabaseClusterFromSnapshot` **should** exist, but doesn't.

It looks like the fix for this is to move the properties `storageEncrypted` and `storageEncryptionKey` at https://github.com/aws/aws-cdk/blob/v1.130.0/packages/@aws-cdk/aws-rds/lib/cluster.ts#L460-L473 from DatabaseClusterProps to DatabaseClusterBaseProps.

### CDK CLI Version

1.130.0

### Framework Version

_No response_

### Node.js Version

_No response_

### OS

_No response_

### Language

Typescript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by relm923**
Closes #17241

Tested by:
```typescript
// 1. Create original cluster with unencrypted storage
new DatabaseCluster(stack, 'Database', {
  engine: DatabaseClusterEngine.AURORA,
  instanceProps: { vpc },
});

// 2. Take snapshot of cluster (mySnapshot)

// 3. Create cluster from snapshot with encrypted storage
new DatabaseClusterFromSnapshot(stack, 'Database', {
  engine: DatabaseClusterEngine.AURORA,
  instanceProps: { vpc },
  snapshotIdentifier: 'mySnapshot',
  storageEncryptionKey: new kms.Key(stack, 'Key'),
});

// 4. Verify new cluster has encrypted storage
```

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-rds/test/cluster.test.ts [hunk 1]
```diff
@@ -1954,6 +1954,31 @@ describe('cluster', () => {
...
+    const vpc = new ec2.Vpc(stack, 'VPC');
+
+    // WHEN
+    new DatabaseClusterFromSnapshot(stack, 'Database', {
+      engine: DatabaseClusterEngine.aurora({ version: AuroraEngineVersion.VER_1_22_2 }),
+      instanceProps: {
+        vpc,
+      },
+      snapshotIdentifier: 'mySnapshot',
+      storageEncryptionKey: new kms.Key(stack, 'Key'),
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/test/cluster.test.ts:1968]: How about using an imported Key in the test (`Key.fromKeyArn()`)? I think that will make the test read nicer.

File: packages/@aws-cdk/aws-rds/test/cluster.test.ts [hunk 2]
```diff
@@ -1954,6 +1954,31 @@ describe('cluster', () => {
...
+    });
+
+    // THEN
+    Template.fromStack(stack).hasResourceProperties('AWS::RDS::DBCluster', {
+      KmsKeyId: {
+        'Fn::GetAtt': [
+          'Key961B73FD',
+          'Arn',
+        ],
+      },
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/test/cluster.test.ts:1978]: Can you also assert here `StorageEncryption` is `true`?

**Review by skinny85 — DISMISSED** [review]
Looks great @relm923, thanks for the contribution!

Two super minor comments on the test before we merge this in 🙂.

File: packages/@aws-cdk/aws-rds/test/cluster.test.ts (hunk 1 above)
Comment (relm923) [reply packages/@aws-cdk/aws-rds/test/cluster.test.ts]: Done

File: packages/@aws-cdk/aws-rds/test/cluster.test.ts (hunk 2 above)
Comment (relm923) [reply packages/@aws-cdk/aws-rds/test/cluster.test.ts:1978]: Done

File: packages/@aws-cdk/aws-rds/test/cluster.test.ts [hunk 3]
```diff
@@ -1954,6 +1954,32 @@ describe('cluster', () => {
...
+    const vpc = new ec2.Vpc(stack, 'VPC');
+
+    // WHEN
+    new DatabaseClusterFromSnapshot(stack, 'Database', {
+      engine: DatabaseClusterEngine.aurora({ version: AuroraEngineVersion.VER_1_22_2 }),
+      instanceProps: {
+        vpc,
+      },
+      snapshotIdentifier: 'mySnapshot',
+      storageEncryptionKey: kms.Key.fromKeyArn(stack, 'Key', 'KeyArn'),
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/test/cluster.test.ts:1968]: Pretty sure you need an actual ARN here 😉. See [here](https://github.com/aws/aws-cdk/blob/19ead77c6470c0aca4900c4ee200a7c4e66ebbed/packages/%40aws-cdk/aws-codepipeline-actions/test/codecommit/codecommit-source-action.test.ts#L533-L534).

File: packages/@aws-cdk/aws-rds/test/cluster.test.ts [hunk 4]
```diff
@@ -1954,6 +1954,32 @@ describe('cluster', () => {
...
+    });
+
+    // THEN
+    Template.fromStack(stack).hasResourceProperties('AWS::RDS::DBCluster', {
+      KmsKeyId: {
+        'Fn::GetAtt': [
+          'Key961B73FD',
+          'Arn',
+        ],
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/test/cluster.test.ts:1977]: Pretty sure this assertions is now incorrect 🙂.

**Review by skinny85 — COMMENTED** [review]
We're almost there 🙂.

**Review by skinny85 — APPROVED** [review]
Thanks for the contribution @relm923!

**Pull request merged**